src/sonarr.ts

```diff
@@ -111,35 +151,165 @@ async function checkExclusionTag(
   const { data: tags, error: tagsError } = await sonarrApi.GET('/api/v3/tag');
   if (tagsError || !tags) {
     throw {
-      message: `Failed to get tags from Sonarr\n${tagsError}`,
+      message: `Failed to get tags from Sonarr.\n${tagsError}`,
       level: 'error',
     };
   }
 
   if (hasExclusionTag(tags, series.tags)) {
     throw {
-      message: `${episodeString} has an exclusion tag, skipping`,
+      message: `${episodeString} has an exclusion tag, skipping.`,
+      level: 'warn',
+    };
+  }
+}
+
+async function checkSeasonsWatched(series: Series): Promise<void> {
+  if (!series.seasons) {
+    throw {
+      message: `Seasons are not defined for ${series.title}.`,
+      level: 'warn',
+    };
+  }
+
+  const unwatchedSeasons = series.seasons.filter((season) => season.monitored);
+
+  if (unwatchedSeasons.length !== 0) {
+    throw {
+      message: `${series.title} has ${unwatchedSeasons.length} monitored season(s) remaining, skipping series unmonitor.`,
       level: 'warn',
     };
   }
 }
 
+async function checkEpisodesWatched(
+  series: Series,
+  episode: Episode,
+  seasonString: string,
+): Promise<Season> {
+  const foundSeason = findSeason(series, episode, seasonString);
```

Comment:
Can we move this out of this function and just have this function check if all episodes are watched in a particular season?